lib/hammer_cli_foreman/ssh_keys.rb

```diff
@@ -0,0 +1,41 @@
+module HammerCLIForeman
+  class SSHKeys < HammerCLIForeman::Command
+    resource :ssh_keys
+    command_name "ssh-keys"
+    desc _("Manage SSH Keys sources.")
+
+    class ListCommand < HammerCLIForeman::ListCommand
+      output do
+        field :id, _("Id")
+        field :name, _("Name")
+        field :fingerprint, _("Fingerprint")
+        field :length, _("Length")
+        field :created_at, _("Created at"), Fields::Date
+      end
+
+      build_options
+    end
+
+    class CreateCommand < HammerCLIForeman::CreateCommand
+      command_name "add"
+
+      option "--key-file", "KEY", _("Path to a SSH public key"),
+             :attribute_name => :option_key,
```

Comment:
I found one tricky thing. When you map two options to the same attribute name the latter attribute uses the reader that was already defined. This prevents from using two options with different formatters pointing to the same `:attribute_name`. You'll have to keep the attribute names separate and handle it in `request_params`.

Something like this should fix it:
```ruby
def request_params
  params = super
  params['ssh_key']['key'] ||= options['option_key_file']
  params
end
```